tree-sitter/src/TreeSitter/Unmarshal.hs

```diff
@@ -260,48 +298,67 @@ instance GUnmarshal U1 where
 
 
 -- For unary products:
-instance (Selector s, Unmarshal k) => GUnmarshal (M1 S s (K1 c k)) where
-  gunmarshalNode _ = push getFields >>= gunmarshalProductNode . fromMaybe Map.empty
+instance UnmarshalAnn k => GUnmarshal (K1 c k) where
+  gunmarshalNode node = K1 <$> unmarshalAnn node
 
--- For sum datatypes:
-instance (GUnmarshalSum f, GUnmarshalSum g, SymbolMatching f, SymbolMatching g) => GUnmarshal (f :+: g) where
-  gunmarshalNode = gunmarshalSumNode @(f :+: g)
+-- For anonymous leaf nodes
+instance GUnmarshal Par1 where
+  gunmarshalNode node = Par1 <$> unmarshalAnn node
+
+instance (Unmarshal t, SymbolMatching t) => GUnmarshal (Rec1 t) where
+  gunmarshalNode node = Rec1 <$> unmarshalNode node
 
 -- For product datatypes:
 instance (GUnmarshalProduct f, GUnmarshalProduct g) => GUnmarshal (f :*: g) where
-  gunmarshalNode _ = push getFields >>= gunmarshalProductNode @(f :*: g) . fromMaybe Map.empty
-
-class GUnmarshalSum f where
```

Comment:
Happily, `GUnmarshalSum` is no longer necessary _at all_. We just use `GUnmarshal` to handle sums.